app/governance/proposal/page.tsx

```diff
@@ -0,0 +1,315 @@
+"use client";
+
+import { useRouter, useSearchParams } from "next/navigation";
+import Text from "@/components/text";
+import styles from "./proposalModal.module.scss";
+import {
+  calculateVotePercentages,
+  formatDeposit,
+  formatProposalStatus,
+  formatProposalType,
+  formatTime,
+} from "@/utils/gov/formatData";
+import Icon from "@/components/icon/icon";
+import Button from "@/components/button/button";
+import useProposals from "@/hooks/gov/useProposals";
+import { useState } from "react";
+import { Proposal } from "@/hooks/gov/interfaces/proposal";
+import useCantoSigner from "@/hooks/helpers/useCantoSigner";
+import Splash from "@/components/splash/splash";
+import { VoteOption } from "@/hooks/gov/interfaces/voteOptions";
+import Image from "next/image";
+import { NEW_ERROR } from "@/config/interfaces";
+
+import { VotingInfoBox } from "../components/VotingInfoBox/VotingInfoBox";
+import { TransactionFlowType } from "@/transactions/flows/flowMap";
+import { NewTransactionFlow } from "@/transactions/flows/types";
+import Spacer from "@/components/layout/spacer";
+
+export default function Page() {
+  function castVote(proposalId: number, voteOption: VoteOption | null) {
+    if (signer) {
+      if (!voteOption) {
+        return NEW_ERROR("Please select a vote option");
+      }
+      //console.log("cast vote test");
+      const newFlow: NewTransactionFlow = {
+        icon: "/canto.svg",
+        txType: TransactionFlowType.VOTE_TX,
+        title: "Voting",
+        params: {
+          chainId: chainId,
+          ethAccount: signer?.account.address ?? "",
+          proposalId: proposalId,
+          voteOption: voteOption,
+        },
+      };
+      txStore?.addNewFlow({
+        txFlow: newFlow,
+        ethAccount: signer.account.address,
+      });
+    }
+  }
+
+  const searchParams = useSearchParams();
+  const id = searchParams.get("id");
+  const proposalId = Number(id);
+  const router = useRouter();
+
+  const { txStore, signer, chainId } = useCantoSigner();
+  const { proposals, isProposalsLoading } = useProposals({ chainId: chainId });
```

Comment:
do we really need all of the proposals if we are on a single page for one proposal only?

Reply:
We are not actually using singleProposal API as it was not working at that time. Will look into this after Refactoring.